tests/integration/gha-run.sh

```diff
@@ -31,6 +31,9 @@ function login_azure() {
 }
 
 function create_cluster() {
+    # First, ensure that the cluster didn't fail to get cleaned up from a previous run.
+    az aks delete -g "kataCI" -n "$(_print_cluster_name)" -y || true
```

Comment:
@sprt, I like the approach, but we need to take into consideration that:
1. Sometimes we hit this because the cluster is in the process of being deleted
2. Deleting a cluster may take several minutes

Does `az aks delete` blocks till the cluster is fully deleted?  If not, we must busy-wait till the cluster is actually removed.

Reply:
`delete_cluster` passes `--no-wait` which makes it non-blocking, the default should be a blocking wait.

on that note: let's extend `delete_cluster` to support passing wait/nowait and call it from here.

Reply:
Okay, then I'd be happy to get this one in with @jepio's suggestion addressed.

Thanks for the explanations @jepio and @Ji-Xinyou!

Reply:
Thanks for pointing this out. As discussed offline with Fabiano, I removed `--no-wait` from `delete_cluster()` so that subsequent runs would wait for deletion of the previous cluster. I also implemented Jeremi's suggestion to just call `delete_cluster()` here.